examples/gopherjs-humble/go/views/todo.go

```diff
@@ -0,0 +1,106 @@
+package views
+
+import (
+	"strings"
+
+	"github.com/go-humble/examples/todomvc/go/models"
+	"github.com/go-humble/examples/todomvc/go/templates"
+	"github.com/go-humble/temple/temple"
+	"github.com/go-humble/view"
+	"honnef.co/go/js/dom"
+)
+
+var (
+	todoTmpl = templates.MustGetPartial("todo")
+)
+
+// Todo is a view for a single todo item.
+type Todo struct {
+	Model *models.Todo
+	tmpl  *temple.Partial
+	view.DefaultView
+}
+
+// NewTodo creates and returns a new Todo view, using the given todo as the
+// model.
+func NewTodo(todo *models.Todo) *Todo {
+	return &Todo{
+		Model: todo,
+		tmpl:  todoTmpl,
+	}
+}
+
+// Render renders the Todo view and satisfies the view.View interface.
+func (v *Todo) Render() error {
+	if err := v.tmpl.ExecuteEl(v.Element(), v.Model); err != nil {
+		return err
+	}
+	v.delegateEvents()
+	return nil
+}
+
+// delegateEvents adds all the needed event listeners to the Todo view.
+func (v *Todo) delegateEvents() {
+	view.AddEventListener(v, "click", ".toggle", v.Toggle)
+	view.AddEventListener(v, "click", ".destroy", v.Remove)
+	view.AddEventListener(v, "dblclick", "label", v.Edit)
+	view.AddEventListener(v, "blur", ".edit", v.CommitEdit)
+	view.AddEventListener(v, "keypress", ".edit",
+		triggerOnKeyCode(enterKey, v.CommitEdit))
+	view.AddEventListener(v, "keydown", ".edit",
+		triggerOnKeyCode(escapeKey, v.CancelEdit))
+
+}
+
+// Toggle toggles the completeness of the todo.
+func (v *Todo) Toggle(ev dom.Event) {
+	v.Model.Toggle()
+}
+
+// Remove removes the todo form the list.
+func (v *Todo) Remove(ev dom.Event) {
+	v.Model.Remove()
+}
+
+// Edit puts the Todo view into an editing state, changing it's appearance and
+// allowing it to be edited.
+func (v *Todo) Edit(ev dom.Event) {
+	addClass(v.Element(), "editing")
+	input, ok := v.Element().QuerySelector(".edit").(*dom.HTMLInputElement)
+	if !ok {
+		panic("Could not convert to dom.HTMLInputElement")
```

Comment:
Is this manual handling of failed type assertions really needed here? I think you could just replace all instances of:

```Go
foo, ok := x.(y)
if !ok {
    panic("Could not convert x to y")
}
```

With the equivalent shorter version:

```Go
foo := x.(y)
```

See https://golang.org/ref/spec#Type_assertions:

> If the type assertion is false, a run-time panic occurs.

The compiler will produce pretty much the same panic message as what you've written manually. E.g.:

```
panic: interface conversion: foo is bar, not *dom.HTMLInputElement
```

Unless there's a specific reason you're choosing to go with the long form, perhaps something related to GopherJS?